internal/services/cosmos/cosmosdb_account_resource_test.go

```diff
@@ -978,6 +979,39 @@ func TestAccCosmosDBAccount_localAuthenticationDisabled(t *testing.T) {
 	})
 }
 
+func TestAccCosmosDBAccount_defaultCreateMode(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_cosmosdb_account", "test")
+	r := CosmosDBAccountResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.defaultCreateMode(data, documentdb.DatabaseAccountKindGlobalDocumentDB, documentdb.DefaultConsistencyLevelEventual),
+			Check: acceptance.ComposeAggregateTestCheckFunc(
+				checkAccCosmosDBAccount_basic(data, documentdb.DefaultConsistencyLevelEventual, 1),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccCosmosDBAccount_restoreCreateMode(t *testing.T) {
+	if ok := skipRestoreCreateMode(); ok {
+		t.Skip("Skipping as one of `ARM_TEST_SOURCE_COSMOSDB_ACCOUNT_ID`, `ARM_TEST_RESTORE_TIMESTAMP_IN_UTC`, `ARM_TEST_DATABASE_NAME` AND `ARM_TEST_COLLECTION_NAME` was not specified")
+	}
```

Comment:
how come we are not creating a source database during the test, and then restoring from it?

Reply:
Based on below two reasons, I assume we cannot create a source database during the test.

**Reason 1:**
Below is the description of restorableDatabase. Per the description, it's not a proper TF candidate. So we cannot refer the resource id of restorableDatabase in TF.
```
The restorable database account is not a tracked resource because any database account with `continuousModeBackupPolicy` (live account or accounts deleted in last 30 days) are the restorable database accounts. So there cannot be Create/Update/Delete operations on the restorable database accounts. They can only be read.
```

**Reason 2:**
API requires the resource id of restorableDatabase and the timestamp of database or collection. Below is the example. However, the segment value of restorableDatabaseAccounts is a guid not a name and the restore timestamp value must be the creation time of database or collection. So these two values have to be manually retrieved by "[Restorable Database Accounts Get/List](https://docs.microsoft.com/en-us/rest/api/cosmos-db-resource-provider/2021-04-01-preview/restorable-database-accounts)" API.
```
source_cosmosdb_account_id = "/subscriptions/xx-xx-xx-xx-xx/providers/Microsoft.DocumentDB/locations/eastus/restorableDatabaseAccounts/xx-xx-xx-xx"
restore_timestamp_in_utc = "2021-11-26T02:24:43Z"
```

Reply:
for 1 - if i understand correctly you can only restore from deleted cosmosdb accounts? what about restoring from a `TypePeriodic` backup? so could you not create the account, wait, save the id, somewhere, delete it and then attempt a restore?
for 2 - there must be an api to get restore timestamps then? maybe we need a resource for users so theyc an pass in an account ID to a data source to get a list of valid timestamps? otherwise how do you know what is a valid timestamp from tf? what are valid timestamps for continues? anytime? periodic?

Reply:
for 1- we only can restore from any cosmosdb accounts with `TypeContinuous`. Restoring doesn't support `TypePeriodic`. We can create the cosmos db account and save the resource id in tc. But the resource id is useless to us/testcase because the resource id  of cosmos db account is different with the resource id of the restorable account id. The resource id of the restorable account id must be retrieved by api https://docs.microsoft.com/en-us/rest/api/cosmos-db-resource-provider/2021-04-01-preview/restorable-database-accounts/list. So you mean we have to add a Computed property to expose the value of the restorable account id?
The resource id of cosmosdb account:
```
/subscriptions/xx-xx-xx-xx/resourceGroups/acctestRG-cosmos-test19/providers/Microsoft.DocumentDB/databaseAccounts/acctest-ca-test19
```
The resource id of the restorable account id:
```
/subscriptions/xx-xx-xx-xx/providers/Microsoft.DocumentDB/locations/eastus/restorableDatabaseAccounts/70570dae-4eec-4537-83bd-7dee820faefe
```

Reply:
@katbyte , thanks for your suggestion. I've added data source and updated test case. Please take another look. Thanks.